tools/debugging/extract_sp_exceptions.sh

```diff
@@ -42,94 +40,81 @@ else
     done
 
-    latest_sp_log=$(${CURL_COMMAND} "${current_server}${scenario}" | grep '^<a ' | cut -d\" -f2 | grep ".log.gz$" | sort | tail -n 1)
-    ${WGET_DIR} -q -P "${DESTINATION_DIR}" "${current_server}${scenario}${latest_sp_log}" &
-
-    wait
+    scp "${sources[@]}" "${DESTINATION_DIR}"
 }
 
 function download_server_logs {
     current_server=$1
-    scenarios=$(${CURL_COMMAND} "${current_server}" | grep '^<a ' | cut -d\" -f2)
-    for scenario in $scenarios; do
-        run_number=$(${CURL_COMMAND} "${current_server}${scenario}run_number.txt")
-        [ -n "$run_number" ] || { echo 'Could not find run number!'; exit 1; }
-        download_nodes_logs "$scenario" "${current_server}" "${run_number}"
-        echo -e "\t - ${scenario}, run_number: ${run_number}"
-    done;
 
-    separator
+    (
+        cd "${SCENARIOS_DIR}"
+
+        ssh "$current_server" '
+        cd /var/lib/scenario-player;
+        source env.sh
+
+        cd /data/scenario-player/scenarios;
+        for scenario_path in "$SCENARIOS_DIR"/ci/"$SP"/*.yaml; do
+            scenario_file=$(basename "$scenario_path")
+            scenario=${scenario_file//.yaml}
+
+            run=$(cat "${scenario}"/run_number.txt)
+
+            find "$scenario"/node_"$run"_*
+            ls "$scenario"/scenario-player-run* -1 -t | head -1
+        done | tar zcf - -T -
+        ' | tar zxf -
+    )
 }
 
 function search_for_failures {
     mkdir -p "${DESTINATION_DIR}/errors/"
     echo -e "${BOLD}Looking for failures${RESET}"
-    scenarios_dir="${DESTINATION_DIR}/scenarios"
-
-    for scenario_dir in "${scenarios_dir}"/*; do
-        scenario=$(basename "${scenario_dir}")
-        print_bold "${scenario}"
-        separator
-
-        for node_dir in $(find "${scenario_dir}" -maxdepth 1 -type d -name "node_*"); do
-            result=$(gunzip -c ${node_dir}/*.log.gz | jq --tab 'select (.error!=null or .exception!=null)')
-            if [[ $result == "" ]]; then
-                result=$(cat "${node_dir}"/*.stderr | grep -v Starting | grep -v Stopped)
-            fi
-            if [[ $result != "" ]]; then
-                print_bold "- Found error in ${node_dir}"
-                echo -e "${result}"
-                echo "${result}" > "${DESTINATION_DIR}/errors/${scenario}.node.log.gz"
-            fi
-        done
-
-        separator
-        sp_error=$(gunzip -c "${scenario_dir}"/scenario-player-run_*.log.gz | jq 'select (.error!=null or .exception!=null)')
-
-        if [[ $sp_error != "" ]]; then
-            print_bold "- SP reported an error in ${scenario_dir}"
-
-            echo -e "${sp_error}"
-            echo "${sp_error}" > "${DESTINATION_DIR}/errors/${scenario}.sp.log"
-        fi
 
-        separator
+    for scenario_logs in $(find "${SCENARIOS_DIR}" -maxdepth 2 -type f -name "scenario-player*.gz"); do
+        scenario_successful=$(gunzip -c  "$scenario_logs" | jq 'select(.result == "success")')
+
+        if [ -z "${scenario_successful}" ]; then
```

Comment:
Not all exceptions are failures, namely, some scenarios will test race conditions, which raise a recoverable exception.

This makes sure only the scenarios that have failed actually print anything to the screen.